HW2/P2/P2.py

```diff
@@ -40,12 +40,12 @@
     # You should explore different values for the number of locks in the medium
     # grained locking
     ########################################
-    N = 10
-    counts[:] = orig_counts
-    with Timer() as t:
-        move_data_medium_grained(counts, src, dest, 100, N)
-    assert counts.sum() == total, "Wrong total after move_data_medium_grained"
-    print("Medium grained uncorrelated: {} seconds".format(t.interval))
+    for N in range(1,20,1):
```

Comment:
I don't know if this is a sufficient enough range for N, you should look into going into the 100s and 1000s for a better idea of what's going on.